catalog/transformer/catalog-transformer-streaming-impl/src/main/java/org/codice/ddf/transformer/xml/streaming/impl/Gml3ToWktImpl.java

```diff
@@ -106,13 +120,97 @@ public Object parseXml(InputStream xml) throws ValidationException {
     }
   }
 
-  private Geometry convertCRS(Geometry geometry) throws ValidationException, TransformException {
-    return JTS.transform(geometry, getLatLonTransform());
+  /**
+   * Iterate over the geometry collection and convert the non-standard LinearRing to LineString
+   *
+   * @param geometryCollection the geometryCollection to be iterated
+   * @return an edited geometryCollection with LineString in place of LinearRing
+   */
+  private GeometryCollection standardizeGeometryCollection(GeometryCollection geometryCollection) {
+    ArrayList<Geometry> geometries = new ArrayList<>();
+
+    for (int i = 0; i < geometryCollection.getNumGeometries(); i++) {
+      Geometry geometry = standardizeGeometry(geometryCollection.getGeometryN(i));
+      geometries.add(geometry);
+    }
+
+    if (geometryCollection.getClass() == MultiPoint.class) {
```

Comment:
❗️ Should use `instanceof` instead of equals on `getClass` so that extended JTS classes will still match.  This violates the Liskov substitution principle.

Reply:
Changed to use instanceof.